src/classic/clvm_tools/stages/stage_2/module.rs

```diff
@@ -594,56 +602,61 @@ fn build_macro_lookup_program(
     }
 }
 
+#[allow(clippy::too_many_arguments)]
 fn add_one_function(
     allocator: &mut Allocator,
     args_root_node: &NodePath,
     macro_lookup_program: NodePtr,
     constants_symbol_table: &[(NodePtr, Vec<u8>)],
-    compiled_functions_: HashMap<Vec<u8>, NodePtr>,
+    // Note: mut here means: the body of this function will mutate this by-value
+    // parameter, not that the mutability is visible to the caller.
+    //
+    // My own style generally avoid this, but reviewers tend to dislike having a
+    // phantom name that is copied, so i am using the more brief style here.
+    mut compiled: CompileOutput,
     name: &[u8],
     lambda_expression: NodePtr,
-) -> Result<HashMap<Vec<u8>, NodePtr>, EvalErr> {
-    let mut compiled_functions = compiled_functions_;
-    m! {
-        com_atom <- allocator.new_atom("com".as_bytes());
-        opt_atom <- allocator.new_atom("opt".as_bytes());
+    has_constants_tree: bool,
+) -> Result<CompileOutput, EvalErr> {
```

Comment:
Richard: consider monoid style for CompileOutput: return just new additions to CompileOutput rather than copy and add.